ipaserver/plugins/cert.py

```diff
@@ -860,6 +864,27 @@ def execute(self, csr, all=False, raw=False, chain=False, **kw):
         )
 
 
+def _emails_are_valid(csr_emails, principal_emails):
+    """
+    Checks if any email address from certificate does not
+    appear in ldap entry, comparing the domain part case-insensitively.
+    """
+
+    if not any(principal_emails):
+        return False
+
+    def lower_domain(email):
+        email_splited = email.split('@', 1)
+        email_splited[1] = email_splited[1].lower()
+
+        return '@'.join(email_splited)
```

Comment:
The `join` is OK, but I think not necessary, because you will always be comparing values
returned from `lower_domain`, and lists will compare properly.  So joining is just a bit of
unneeded but harmless work.